03/app.js

```diff
@@ -1,5 +1,29 @@
-document.addEventListener('DOMContentLoaded', init);
+document.addEventListener("DOMContentLoaded", init);
+const btnEl = document.querySelector("button");
+const spanEl = document.querySelector("span");
 
 function init() {
-    console.log('DOM');
-}
\ No newline at end of file
+  console.log("DOM");
+  loadIp();
+}
+
+function loadIp() {
+  fetch("https://api.ipify.org?format=json")
+    .then((resp) => {
+      if (resp.ok) {
+        return resp.json();
+      }
+      return Promise.reject(resp);
+    })
+    .then((ip) => insertIp(ip))
+    .catch((err) => console.error(err))
+    .finally(() => {
+      console.log("Api zakończyło działanie");
+    });
+}
+
+function insertIp(ip) {
```

Comment:
Raczej powinna być inna kolejność (odwrotna). Jeśli ktoś kliknie w przycisk to wysyłamy zapytanie do API i jak otrzymamy odpowiedź to ją osadzamy w HTML. Obecne rozwiązanie może powodować niepotrzebne problemy np. ktoś kliknie w przycisk, ale nie pobierzemy jeszcze danych z API i to nie zadziała.